pkg/steps/template.go

```diff
@@ -856,20 +861,32 @@ func getContainersMap(containers string) map[string]struct{} {
 	return c
 }
 
-func outputAnnotatedContainerLogs(podClient coreclientset.PodInterface, pod *coreapi.Pod) error {
+func artifactsAnnotatedContainerLogs(podClient coreclientset.PodInterface, pod *coreapi.Pod, artifactDir string) error {
+	var validationErrors []error
 	statuses := gatherContainerStatuses(pod)
 	containersToOutput := getContainersMap(pod.ObjectMeta.Annotations[showContainerOutputAnnotation])
+
 	for _, status := range statuses {
 		if s := status.State.Terminated; s != nil {
 			if _, exists := containersToOutput[status.Name]; exists {
-				log.Printf("Container %s logs:", status.Name)
-				if err := printContainerLogsToStdout(podClient, status.Name, pod.Name); err != nil {
-					return fmt.Errorf("%v", err)
+				file, err := os.Create(fmt.Sprintf("%s/%s/%s-out.log", artifactDir, pod.Name, status.Name))
+				if err != nil {
+					validationErrors = append(validationErrors, fmt.Errorf("Cannot create file: %v", err))
```

Comment:
Is there any reason for not skipping the rest of the body here? What is the point of continuing?

Reply:
Yeah, in the next cycle iteration, but why executing the rest of the body? The `Copy` there will fail no matter what.

Reply:
Yes, the intention here is output the errors if there are also for the other containers.

Reply:
Other containers will be handled by different iterations of the cycle. I am asking why to continue executing a single iteration when a file creation fails.

Reply:
This still holds.